### DISCUSSION TITLE
Go test explorer

### DISCUSSION THREAD

**Discussion by tradiff** [JanDeDobbeleer/oh-my-posh#313] [Troubleshoot] [answered]
I saw that a VSCode extensions file was added, which includes recommendations for "hbenl.vscode-test-explorer" and  "ethan-reesor.vscode-go-test-adapter".  I have installed both, but for the life of me, I cannot get this test explorer to display any tests.  I looked for documentation on both projects, and didn't find much.  I've also tried uninstalling/reinstalling both extensions, and opening the OMP directory, or opening the src directory.  Does anyone have any idea what I might be doing wrong?  @lnu perhaps

**Comment by lnu** [comment]
When I debug on wsl I open the root folder(opening src only also works) and only use the recommended extensions:
![image height](https://user-images.githubusercontent.com/1829553/103509543-add3cd80-4e63-11eb-96a4-f1724b1aa20d.png)
and the test appears automatically:
![image](https://user-images.githubusercontent.com/1829553/103509572-b88e6280-4e63-11eb-8397-5a9a17ba350d.png)
Do you see the run/debug options?
![image](https://user-images.githubusercontent.com/1829553/103509945-80d3ea80-4e64-11eb-8ccd-28906944ce4e.png)

**Reply by tradiff** [reply]
Thanks.  The only thing I really spotted that we're doing differently is you're using WSL, and I'm using Windows.  I tried WSL, and it still didn't load the tests.  As far as whether I see the actions at the top of the functions, I only see the two that are provided by the Go extension. Nothing provided by "Go Test Explorer".  (so I only see "run test" and "debug test").

I'm still not able to get this working, but I saw [this issue](https://gitlab.com/firelizzard/vscode-go-test-adapter/-/issues/11) which I've subscribed to, so maybe that'll end up paying off. For now it's not that big of a deal, I run tests using the CLI.  IDE integration would be cool though.  I'll report back here if I find any solution.

**Comment by lnu** [comment]
Try also to enable the debug output to see what's happening:
![image](https://user-images.githubusercontent.com/1829553/103510195-0061b980-4e65-11eb-8c00-d0a4577a4f4f.png)
![image](https://user-images.githubusercontent.com/1829553/103511040-95b17d80-4e66-11eb-8175-467142f74250.png)

**Comment by lnu** [comment]
@JanDeDobbeleer does the test explorer work for you?

**Reply by JanDeDobbeleer** [reply]
sorry, missed this. I can see this in macOS yes.

**Reply by JanDeDobbeleer** [reply]
<img width="273" alt="image" src="https://user-images.githubusercontent.com/2492783/103892309-aaeb0e00-50eb-11eb-967d-804afd82ac83.png">

**Comment by lnu** [comment]
@TravisTX any follow up on this one?

**Reply by tradiff** [reply]
Nope. I went ahead and posted a reply in the [vscode-go-test-adapter thread](https://gitlab.com/firelizzard/vscode-go-test-adapter/-/issues/11) with some more information. Hopefully the maintainer of that project has some ideas.

**Comment by tradiff** [comment, marked as answer]
This is resolved for me now. In case anyone else experiences this issue, I had to enable the "Use Language Server" option in the VS settings for the "Go" extension:

![image](https://user-images.githubusercontent.com/934490/104853370-9d890d00-58bd-11eb-8862-faf00066c9a8.png)

According to the maintainer, this should be fixed without this feature in the next version, as explained here  https://gitlab.com/firelizzard/vscode-go-test-adapter/-/issues/11#note_487271987

**Reply by lnu** [reply]
I missed that one. I don’t remember if I set it manually or not.

**Reply by JanDeDobbeleer** [reply]
I did set it myself I think.
